docs/structured-streaming-programming-guide.md

```diff
@@ -1814,6 +1814,23 @@ Specifically for built-in HDFS state store provider, users can check the state s
 it is best if cache missing count is minimized that means Spark won't waste too much time on loading checkpointed state.
 User can increase Spark locality waiting configurations to avoid loading state store providers in different executors across batches.
 
+### RocksDB state store implementation
```

Comment:
Actually we don't mention state store clearly in this doc before. The doc only mentions `spark.sql.streaming.stateStore.providerClass` could be used to set up state store implementation.

How about introducing both state stores? E.g.,

 ``` markdown
### State Store
#### HDFS state store provider
.....
#### RocksDB state store provider
.....
```

Reply:
Make sense, done in dec0bae